tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -127,16 +127,16 @@ def montecarlo_main_loop(
     """
     output_nus = np.empty_like(packet_collection.packets_output_nu)
     last_interaction_types = (
-        np.ones_like(packet_collection.packets_output_nu) * -1
+        np.ones_like(packet_collection.packets_output_nu).astype(np.int64) * -1
```

Comment:
You can use the `dtype=np.int64` keyword argument in the array definition, instead of recasting the array to int64.

Reply:
Sure understood ... I wanted to know that if I use`dtype=np.int64` & if I used `astype(np.int64)` what is the main difference that may occur in the execution of the loop & hence generation of results? 🤔

Reply:
Honestly, I don't know! I recommend investigating yourself. I suggest the `dtype` kwarg because it is part of the array creation rather than acted upon the array, so there may be an order-of-operations performance impact.

Reply:
Sure, I will check it out & convey it here too 😄 ... went with the `astype` modification as the variables [here](https://github.com/tardis-sn/tardis/pull/1420/commits/81d530712d3fc8201766355f6ce155e8de974e34#diff-fd672eb0f9f2329112e41aa8908ab882a00c598a87719558899ae5751757d92fL198-L200) so as to follow the same pattern 😓

Reply:
So on that line, there is not an array being created, so they have to be recast to a new data type using `astype`. When an array is created in numpy, you can specify the `dtype `at creation time.

Reply:
Awesome, I had read about this 😄 ... Thank You very much for clarifying this ... Makes a lot of sense :)

Reply:
It's ready now!! ... please do let me know anything I can help with :)

Reply:
Hello, 
I found [*this*](https://stackoverflow.com/a/39626328/3714343) & does look like a complete good explanation for the difference